osf/migrations/0069_auto_20171127_1119.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.7 on 2017-11-27 17:19
+from __future__ import unicode_literals
+import logging
+
+from django.db import migrations
+from osf.models import PreprintService
+logger = logging.getLogger(__name__)
+
+def add_preprint_doi_created(apps, schema_editor):
+    """
+    Sets preprint_doi_created equal to date_published for existing published preprints.
+    """
+    null_preprint_doi_created = PreprintService.objects.filter(preprint_doi_created__isnull=True, date_published__isnull=False)
+    preprints_count = null_preprint_doi_created.count()
+    current_preprint = 0
+    logger.info('{} published preprints found with preprint_doi_created is null.'.format(preprints_count))
+
+    for preprint in null_preprint_doi_created:
```

Comment:
Rather than looping and doing n queries, you should be able to do one query:

```python
PreprintService.objects.filter(...).update(preprint_doi_created=F('date_published')
```

Reply:
@sloria, I wanted to check whether the DOI had actually been created before populating the preprint_doi_created field, which was the main reason for the loop.

Reply:
Ah ok.